### PR TITLE
Client auth flow: pass maxTimeToLive (FOLLOW-136)

### PR DESCRIPTION
**Pull request by nomeata**
We were not passing `maxTimeToLive` through to the backend. This fixes
this.

It also refactors the selenium test suite mildly and adds tests that
exercise setting the default maxTimeToLive, 1min, 1day and 1 month.

This needs a released update in `auth-client` for the tests to work, so
we’ll have to wait for https://github.com/dfinity/agent-js/pull/417

### CODE REVIEW

**Comment by nomeata** [comment]
Harsh review of my TypeScript code is welcome :-)

What is the difference between `bigint` and `BigInt` in typescript? Should we indeed be using both?

File: demos/sample-javascript/src/main.js [hunk 1]
```diff
@@ -8,27 +9,54 @@ const hostUrlEl = document.getElementById("hostUrl");
...
   authClient = await AuthClient.create();
-  principalEl.innerText = await authClient.getIdentity().getPrincipal();
 
-  // Redirect to the identity provider
+  const updateView = () => {
+    const identity = authClient.getIdentity();
+
+    principalEl.innerText = identity.getPrincipal();
+    if (identity instanceof DelegationIdentity) {
+      delegationEl.innerText = JSON.stringify(identity.getDelegation().toJSON());
```
Comment (kritzcreek) [inline demos/sample-javascript/src/main.js:27]:
```suggestion
      delegationEl.innerText = JSON.stringify(identity.getDelegation().toJSON(), undefined, 2);
```

If you'd like it to pretty print the JSON with some spacing and newlines

File: demos/sample-javascript/src/index.html [hunk 2]
```diff
@@ -13,10 +13,24 @@ <h2>Sign In</h2>
...
+	  placeholder="(use default)"
+        />
+      </div>
       <button id="signinBtn">Sign In</button>
       <button id="signoutBtn">Sign Out</button>
-      <h2>Principal:</h2>
+      <h3>Principal:</h3>
       <div id="principal"></div>
+      <h3>Delegation:</h3>
+      <div id="delegation"></div>
```
Comment (kritzcreek) [inline demos/sample-javascript/src/index.html:31]: I'd maybe make this a `pre` (so it's monospaced)

File: demos/sample-javascript/src/main.js [hunk 3]
```diff
@@ -8,27 +9,54 @@ const hostUrlEl = document.getElementById("hostUrl");
...
+
+      const nextExpiration =
+        identity.getDelegation().delegations
+         .map(d => d.delegation.expiration)
+         .reduce((current, next) => next < current ? next : current);
```
Comment (kritzcreek) [inline demos/sample-javascript/src/main.js:32]:
```suggestion
        Math.min(...identity.getDelegation().delegations.map(d => d.delegation.expiration));
```

Might be a little easier for the non FP folks

File: src/frontend/src/selenium.test.ts [hunk 4]
```diff
@@ -387,6 +416,30 @@ async function run_in_browser_common(
...
+  const elem = await driver.findElement(By.id(id));
+  elem.sendKeys(Key.CONTROL + "a");
+  elem.sendKeys(Key.DELETE);
+  elem.sendKeys(text);
+}
+
+async function switchToPopup(driver: ThenableWebDriver) {
+  let handles = await driver.getAllWindowHandles();
+  expect(handles.length).toBe(2);
+  await driver.switchTo().window(handles[1]);
```
Comment (kritzcreek) [inline src/frontend/src/selenium.test.ts:429]: Is this order always deterministic? Could we maybe do something like finding the popup via its window title?

**Review by kritzcreek — CHANGES_REQUESTED** [review]
Looks great! `BigInt` is the constructor, while `bigint` is the type... You can also use `BigInt` as the type, but that causes problems with inference so just stick to `bigint` for types. Hans explained that to me at some point, but I don't remember the details.

File: demos/sample-javascript/src/main.js (hunk 3 above)
Comment (nomeata) [reply demos/sample-javascript/src/main.js:32]: I actually copied that from the `dfinity_wallet` code :-)

File: src/frontend/src/selenium.test.ts (hunk 4 above)
Comment (nomeata) [reply src/frontend/src/selenium.test.ts:429]: So far it works 🤷🏻

**Comment by nomeata** [comment]
I found an existing use of `BigInt` in a type, and replaced that with `bigint`, and thus could remove one conversion. hope that’s fine!

File: src/frontend/src/selenium.test.ts (hunk 4 above)
Comment (kritzcreek) [reply src/frontend/src/selenium.test.ts:429]: Allright, good enough for me. If it ever starts failing the test will fail local enough, that we can figure it out.

**Review by kritzcreek — APPROVED** [review]

**Pull request merged**